common4j/src/main/com/microsoft/identity/common/java/controllers/CommandDispatcher.java

```diff
@@ -222,6 +287,19 @@ public static FinalizableResultFuture<CommandResult> submitSilentReturningFuture
         // set correlation id on parameters as it may not already be set
         commandParameters.setCorrelationId(correlationId);
 
+        final boolean isDeviceCodeFlowRequest = isDeviceCodeFlowRequest(command);
+        // Use DCF thread pool for DCF requests so that in case user chooses to follow interactive sign in,
+        // future silent calls are not blocked
+        if (isDeviceCodeFlowRequest) {
+            commandExecutor = sDCFExecutor;
+            codeMarkerManager.markCode(ACQUIRE_TOKEN_DCF_START);
+            methodName = ":submitDCF";
+        } else {
+            commandExecutor = sSilentExecutor;
+            codeMarkerManager.markCode(ACQUIRE_TOKEN_SILENT_START);
+            methodName = ":submitSilent";
+        }
```

Comment:
Could you pass this as parameter instead of if/else?
Executor.

It's better to keep tag same as methodName only. We can correlate logs based on logs emitted in caller methods.

Reply:
currently there are multiple markers used in this method based on dcf/non-dcf flow, namely 
1. ACQUIRE_TOKEN_DCF_START/ ACQUIRE_TOKEN_SILENT_START
2. ACQUIRE_TOKEN_DCF_EXECUTOR_START/ ACQUIRE_TOKEN_SILENT_EXECUTOR_START
3. ACQUIRE_TOKEN_DCF_COMMAND_EXECUTION_START/ ACQUIRE_TOKEN_SILENT_COMMAND_EXECUTION_START
4. ACQUIRE_TOKEN_DCF_COMMAND_EXECUTION_END/ACQUIRE_TOKEN_SILENT_COMMAND_EXECUTION_END
5. ACQUIRE_TOKEN_DCF_FUTURE_OBJECT_CREATION_END/ ACQUIRE_TOKEN_SILENT_FUTURE_OBJECT_CREATION_END

As per my opinion the best we can do is just pass a flag in parameter and take care of these but then we are already having a utility method to decide whether its a dcf or not based on the command so i feel current approach is a better way.

Let me know if you have a better suggestion for this

Reply:
Then probably all such customizations should be taken as input based on flow. 

(Will not block on this.)